modules/bidglassBidAdapter.js

```diff
@@ -0,0 +1,185 @@
+        return document.referrer;
+      } else {
+        return null;
+      }
+    };
+    let getOrigins = function() {
+      var ori = [window.location.protocol + '//' + window.location.hostname];
+
+      if (window.location.ancestorOrigins) {
+        for (var i = 0; i < window.location.ancestorOrigins.length; i++) {
+          ori.push(window.location.ancestorOrigins[i]);
+        }
+      } else if (window !== window.top) {
+        // Derive the parent origin
+        var parts = document.referrer.split('/');
+
+        ori.push(parts[0] + '//' + parts[2]);
+
+        if (window.parent !== window.top) {
+          // Additional unknown origins exist
+          ori.push('null');
+        }
+      }
+
+      return ori;
+    };
+
+    utils._each(validBidRequests, function(bid) {
+      bid.sizes = ((utils.isArray(bid.sizes) && utils.isArray(bid.sizes[0])) ? bid.sizes : [bid.sizes]);
+      bid.sizes = bid.sizes.filter(size => utils.isArray(size));
+
+      // Stuff to send: [bid id, sizes, adUnitId]
+      imps.push({
+        bidId: bid.bidId,
+        sizes: bid.sizes,
+        adUnitId: utils.getBidIdParameter('adUnitId', bid.params)
+      });
+    });
+
+    // Stuff to send: page URL
+    const bidReq = {
+      reqId: utils.getUniqueIdentifierStr(),
+      imps: imps,
+      ref: getReferer(),
+      ori: getOrigins()
+    };
+
+    let url = 'https://bid.glass/ad/hb.php?' +
+      `src=$$REPO_AND_VERSION$$`;
+
+    return {
+      method: 'POST',
+      url: url,
+      data: JSON.stringify(bidReq),
+      options: {
+        contentType: 'text/plain',
+        withCredentials: false
+      }
+    }
+  },
+
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * @param {ServerResponse} serverResponse A successful response from the server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function(serverResponse) {
+    // const serverBody  = serverResponse.body;
+    // const headerValue = serverResponse.headers.get('some-response-header');
+
+    const bidResponses = [];
+
+    utils._each(serverResponse.body.bidResponses, function(bid) {
+      bidResponses.push({
+        requestId: bid.requestId,
+        cpm: parseFloat(bid.cpm),
+        width: parseInt(bid.width),
+        height: parseInt(bid.height),
+        creativeId: bid.creativeId,
+        dealId: bid.dealId || null,
+        currency: bid.currency || 'USD',
+        mediaType: bid.mediaType || 'banner',
+        netRevenue: true,
+        ttl: bid.ttl || 10,
+        ad: bid.ad
+      });
+    });
+
+    return bidResponses;
+  },
+
+  /**
+   * Register the user sync pixels which should be dropped after the auction.
+   *
+   * @param {SyncOptions} syncOptions Which user syncs are allowed?
+   * @param {ServerResponse[]} serverResponses List of server's responses.
+   * @return {UserSync[]} The user syncs which should be dropped.
+   */
+  getUserSyncs: function() {},

```

Comment:
the formal params are missing here. Can you please add some code, may be just a log in function definition.

Reply:
Thanks for taking the time to review our adapter! We've made the updates as requested. One question - is there a gulp (or other) command to check the coverage for an individual bidder's unit tests?

A note regarding the empty function definitions - we don't plan on using getUserSyncs, onTimeout, onBidWon, or onSetTargeting to any practical effect in the near future.

Reply:
Thanks for the update. You can use `gulp test-coverage`, and then `gulp view-coverage` to view unit test-coverage of any adaptor file. The view command will open coverage on `http://localhost:1999/`, where you can navigate to your file.

Reply:
Thanks!